.github/workflows/regression-report.yml

```diff
@@ -122,24 +163,219 @@ jobs:
+          CONFIG_LINK="$BASE_URL/$CONFIG_BUNDLE/config.xml"
+          PROJECTS_LINK="$BASE_URL/$CONFIG_BUNDLE/list-of-projects.properties"
+          ./.ci/append-to-github-output.sh "config_link" "$CONFIG_LINK"
+          ./.ci/append-to-github-output.sh "projects_link" "$PROJECTS_LINK"
+
+  handle_generate_config:
+    needs: [ parse_comment, checkout_and_cache ]
+    if: needs.parse_comment.outputs.mode == 'generate_config'
+    runs-on: ubuntu-latest
+    outputs:
+      config_link: ${{ steps.generate_config.outputs.config_link }}
+      projects_link: ${{ steps.generate_config.outputs.projects_link }}
+    steps:
+      - name: Restore cached checkouts
+        uses: actions/cache@v4
+        with:
+          path: |
+            .
+            .ci-temp/checkstyle
+          key: ${{ runner.os }}-checkouts-${{ github.sha }}
+
+      - name: Find input file
+        id: find_input_file
+        run: |
+          INPUT_FILE="${{ needs.parse_comment.outputs.input_file }}"
+          CHECKSTYLE_DIR=".ci-temp/checkstyle"
+          FILENAME=$(basename "$INPUT_FILE")
+
+          FOUND_FILES=$(find "$CHECKSTYLE_DIR" -name "$FILENAME" -type f)
+          if [ "$(echo "$FOUND_FILES" | wc -l)" -gt 1 ]; then
+            echo "Error: Multiple matching files found"
+            exit 1
+          elif [ -z "$FOUND_FILES" ]; then
+            echo "Error: Could not find input file"
+            exit 1
+          fi
+          FOUND_FILE="$FOUND_FILES"
+          ./.ci/append-to-github-output.sh "input_file" "$FOUND_FILE"
+
+      - name: Download DiffTool JAR
+        uses: robinraju/release-downloader@v1.8
+        with:
+          repository: "checkstyle/test-configs"
+          tag: "extractor-2024-08-24"
+          fileName: "extractor-all-2024-08-21.jar"
+
+      - name: Set up Java 21
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '21'
+
+      - name: Generate config
+        id: generate_config
+        env:
+          DEFAULT_PROJECTS_LINK: ${{ env.DEFAULT_PROJECTS_LINK }}
+        run: |
+          FOUND_FILE="${{ steps.find_input_file.outputs.input_file }}"
+          BASE_PATH=".ci-temp/checkstyle"
+          OUTPUT_DIR="generated_configs"
+          mkdir -p "$OUTPUT_DIR"
+
+          echo "Original FOUND_FILE = $FOUND_FILE"
+
+          # Remove .ci-temp/checkstyle/ from the beginning of FOUND_FILE
+          RELATIVE_PATH=${FOUND_FILE#.ci-temp/checkstyle/}
+
+          echo "Processed RELATIVE_PATH = $RELATIVE_PATH"
+
+          # Generate a unique name for the config file
+          CONFIG_FILE_NAME=$(basename "$FOUND_FILE" .java)_config.xml
+          CONFIG_FILE_PATH="$OUTPUT_DIR/$CONFIG_FILE_NAME"
+
+          # Generate a name for the projects file
+          PROJECT_FILE_PATH="$OUTPUT_DIR/projects.properties"
+
+          # Run the extractor
+            java -jar extractor-all-2024-08-21.jar "$BASE_PATH" \
```

Comment:
lets reuse variable for version of jar 
example: https://github.com/checkstyle/checkstyle/blob/d32c27b87650da899104201a82b5220ed6c4234a/.github/workflows/google-java-format.yml#L22-L23

Reply:
done

Reply:
`java -jar extractor-all-${{env.EXTRACTOR_VERSION}}.jar "$BASE_PATH" \`

Reply:
done, but as mentioned mismatch with tag version and jar version.